packages/flutter_tools/test/general.shard/artifacts_test.dart

```diff
@@ -3,138 +3,148 @@
+      expect(
+        artifacts.getArtifactPath(Artifact.flutterTester),
+        fileSystem.path.join('/out', 'android_debug_unopt', 'flutter_tester'),
+      );
+      expect(
+        artifacts.getArtifactPath(Artifact.engineDartSdkPath),
+        fileSystem.path.join('/out', 'host_debug_unopt', 'dart-sdk'),
+      );
+    });
 
-      testUsingContext('getEngineType', () {
-        expect(
-          artifacts.getEngineType(TargetPlatform.android_arm, BuildMode.debug),
-          'android_debug_unopt',
-        );
-        expect(
-          artifacts.getEngineType(TargetPlatform.ios, BuildMode.release),
-          'android_debug_unopt',
-        );
-        expect(
-          artifacts.getEngineType(TargetPlatform.darwin_x64),
-          'android_debug_unopt',
-        );
-      }, overrides: <Type, Generator>{
-        FileSystem: () => memoryFileSystem,
-        ProcessManager: () => FakeProcessManager.any(),
-        Platform: () => FakePlatform(operatingSystem: 'linux'),
-      });
+    testWithoutContext('getEngineType', () {
+      expect(
+        artifacts.getEngineType(TargetPlatform.android_arm, BuildMode.debug),
+        'android_debug_unopt',
+      );
+      expect(
+        artifacts.getEngineType(TargetPlatform.ios, BuildMode.release),
+        'android_debug_unopt',
+      );
+      expect(
+        artifacts.getEngineType(TargetPlatform.darwin_x64),
+        'android_debug_unopt',
+      );
+    });
 
-      testUsingContext('Looks up dart.exe on windows platforms', () async {
-        expect(artifacts.getArtifactPath(Artifact.engineDartBinary), contains('.exe'));
-      }, overrides: <Type, Generator>{
-        FileSystem: () => memoryFileSystem,
-        ProcessManager: () => FakeProcessManager.any(),
-        Platform: () => FakePlatform(operatingSystem: 'windows'),
-      });
+    testWithoutContext('Looks up dart.exe on windows platforms', () async {
+      artifacts = LocalEngineArtifacts(fileSystem.currentDirectory.path,
+        fileSystem.path.join(fileSystem.currentDirectory.path, 'out', 'android_debug_unopt'),
+        fileSystem.path.join(fileSystem.currentDirectory.path, 'out', 'host_debug_unopt'),
+        cache: cache,
+        fileSystem: fileSystem,
+        platform: FakePlatform(operatingSystem: 'windows'),
+        processManager: FakeProcessManager.any(),
+      );
+
+      expect(artifacts.getArtifactPath(Artifact.engineDartBinary), contains('.exe'));
+    });
 
-      testUsingContext('Looks up dart on linux platforms', () async {
-        expect(artifacts.getArtifactPath(Artifact.engineDartBinary), isNot(contains('.exe')));
-      }, overrides: <Type, Generator>{
-        FileSystem: () => memoryFileSystem,
-        ProcessManager: () => FakeProcessManager.any(),
-        Platform: () => FakePlatform(operatingSystem: 'linux'),
-      });
+    testWithoutContext('Looks up dart on linux platforms', () async {
+      expect(artifacts.getArtifactPath(Artifact.engineDartBinary), isNot(contains('.exe')));
```

Comment:
doesn't this need a new `artifacts` with `platform: FakePlatform(operatingSystem: 'linux'),`?

Reply:
The platform already setup in the group is linux, so I reused that one

Reply:
ahh, missed that. lgtm.